examples/synapseml/autologging.md

```diff
@@ -0,0 +1,89 @@
+## MlFlow automatic Logging with SynapseML
+
+[MLflow automatic logging](https://www.mlflow.org/docs/latest/tracking.html#automatic-logging) allows you to log metrics, parameters, and models without the need for explicit log statements.
+SynapseML supports autologging for every model in the library.
+
+Install SynapseML library following this [guidance](https://microsoft.github.io/SynapseML/docs/getting_started/installation/)
+
+To enable autologging for SynapseML:
+
+1. Download this customized [log_model_allowlist file](https://mmlspark.blob.core.windows.net/publicwasb/log_model_allowlist.txt) and put it at a place that your code has access to.
+   For example:
+
+- In Synapse `wasb://<containername>@<accountname>.blob.core.windows.net/PATH_TO_YOUR/log_model_allowlist.txt`
+- In Databricks `/dbfs/FileStore/PATH_TO_YOUR/log_model_allowlist.txt`.
+
+2. Set spark configuration `spark.mlflow.pysparkml.autolog.logModelAllowlistFile` to the path of your `log_model_allowlist.txt` file.
+3. Call `mlflow.pyspark.ml.autolog()` before your training code to enable autologging for all supported models.
+
+Note:
+
+1. If you want to support autologging of PySpark models not present in the log_model_allowlist file, you can add such models to the file.
+2. If you've enabled autologging, then don't write explicit `with mlflow.start_run()` as it might cause multiple runs for one single model or one run for multiple models.
```

Comment:
Autologging won't create multiple runs if users call `with mlflow.start_run()`. We can remove this.